### PR TITLE
refactor: remove note creation modal

### PR DESCRIPTION
**Pull request by ezhil56x**
/claim #427 

- New file will be created without a modal
- By default Untitled will be the name of the new file
- The name of the file depends on the first 20 characters of the first line of the file

https://github.com/user-attachments/assets/59a61c2e-d9a6-4ed5-b82b-75575f6526d5

### CODE REVIEW

File: src/components/File/NewNote.tsx [hunk 1]
```diff
@@ -15,83 +12,45 @@ interface NewNoteComponentProps {
...
-
   const { currentlyOpenFilePath } = useFileContext()
 
-  useEffect(() => {
-    if (!isOpen) {
-      setFileName('')
-      setErrorMessage(null)
-    }
-  }, [isOpen])
+  const createNewFile = async () => {
```
Comment (joseplayero) [inline src/components/File/NewNote.tsx:17]: we can get of the newnote component fully right? this function can just go in the iconssidebar where the icon for new note is pressed and we can fully delete the newnotecomponent

File: src/contexts/FileContext.tsx [hunk 2]
```diff
@@ -220,10 +220,23 @@ export const FileProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
     if (filePath !== null && needToWriteEditorContentToDisk && _editor) {
       const markdownContent = getMarkdown(_editor)
       if (markdownContent !== null) {
+        const firstLine = markdownContent.split('\n')[0]
+        const newFileName = `${firstLine
```
Comment (joseplayero) [inline src/contexts/FileContext.tsx:224]: can this sanitization and file naming code be moved to a separate function please

File: src/contexts/FileContext.tsx [hunk 3]
```diff
@@ -220,10 +220,23 @@ export const FileProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
...
+          .replace(/[^a-zA-Z0-9-_ ]/g, '')
```
[the remark this answers is not in the document]
Comment (joseplayero) [reply src/contexts/FileContext.tsx:225]: also this code will require similar checks to what is done above with getAllFilenamesInDirectory to ensure  files aren't overwritten if they have the same first 20 characters or so

**Review by joseplayero — CHANGES_REQUESTED** [review]

File: src/contexts/FileContext.tsx [hunk 4]
```diff
@@ -220,10 +220,23 @@ export const FileProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
...
+          .trim()
+          .substring(0, 20)}.md`
+        const newFilePath = await window.path.join(await window.path.dirname(filePath), newFileName)
+        await window.fileSystem.renameFileRecursive({
```
Comment (joseplayero) [inline src/contexts/FileContext.tsx:229]: also, renaming the file should only occur once. 
only the first time the file is saved after the user has modified the content. we can determine this if the filename has untitled in it

**Comment by ezhil56x** [comment]
@joseplayero Should I remove the NewNote component fully. If so can I move the functions that are required to create a new note to FileContext?

**Comment by ezhil56x** [comment]
Resolved other changes

**Comment by joseplayero** [comment]
i still see some bugs and places for improvement @ezhil56x 

and i dont have time right now to review properly but will get back to you at the end of the week with a full review. does that sound fine to you?

**Comment by ezhil56x** [comment]
Sounds good

**Comment by joseplayero** [comment]
this become larger refactor so i will take over completing the task.

i have issued the bounty to thank for help @ezhil56x

**Comment by ezhil56x** [comment]
@joseplayero
Can you add `hacktoberfest-accepted` label to this PR. It will help me complete [hacktoberfest](https://hacktoberfest.com/) this year.
Thanks!

**Pull request merged**